testsuite/smoke-test/src/event_fetcher.rs

```diff
@@ -1,63 +1,85 @@
-    client.create_next_account(false).unwrap();
-    client
-        .mint_coins(&["mintb", "0", "100", "XUS"], true)
-        .unwrap();
-
-    client
-        .mint_coins(&["mintb", "1", "100", "XUS"], true)
-        .unwrap();
-
-    client
-        .transfer_coins(&["tb", "0", "1", "3", "XUS"], true)
-        .unwrap();
-
-    client
-        .transfer_coins(&["tb", "0", "1", "4", "XUS"], true)
-        .unwrap();
-
-    let (account, _) = client.get_account_address_from_parameter("0").unwrap();
-    let (sent_handle, received_handle) = runtime
-        .block_on(events_fetcher.get_payment_event_handles(account))
-        .unwrap()
-        .unwrap();
-    let mut sent_events = runtime
-        .block_on(events_fetcher.get_all_events(&sent_handle))
-        .unwrap();
-    let received_events = runtime
-        .block_on(events_fetcher.get_all_events(&received_handle))
-        .unwrap();
-
-    assert_eq!(received_events.len(), 1);
-    assert_eq!(received_handle.count(), 1);
-    assert_eq!(sent_handle.count(), 2);
-    assert_eq!(sent_events.len(), 2);
-    let evt1 = sent_events.pop().unwrap();
-    let evt2 = sent_events.pop().unwrap();
-
-    if let EventDataView::SentPayment { amount, .. } = evt1.data {
-        assert_eq!(amount.amount, 4000000);
-    } else {
-        panic!("invalid event");
+use diem_events_fetcher::DiemEventsFetcher;
+use diem_sdk::{client::views::EventDataView, transaction_builder::Currency};
+use forge::{PublicUsageContext, PublicUsageTest, Result, Test};
+
+pub struct EventFetcher;
+
+impl Test for EventFetcher {
+    fn name(&self) -> &'static str {
+        "smoke-test::event-fetcher"
     }
+}
+
+impl PublicUsageTest for EventFetcher {
+    fn run<'t>(&self, ctx: &mut PublicUsageContext<'t>) -> Result<()> {
+        let runtime = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()?;
+
+        let client = ctx.client();
+        let factory = ctx.transaction_factory();
+
+        let mut account1 = ctx.random_account();
+        ctx.create_parent_vasp_account(account1.authentication_key())?;
+        ctx.fund(account1.address(), 100_000_000)?;
+
+        let account2 = ctx.random_account();
+        ctx.create_parent_vasp_account(account2.authentication_key())?;
+        ctx.fund(account2.address(), 100_000_000)?;
+
+        let txn = account1.sign_with_transaction_builder(factory.peer_to_peer(
+            Currency::XUS,
+            account2.address(),
+            3_000_000,
+        ));
+        client.submit(&txn)?;
+        client.wait_for_signed_transaction(&txn, None, None)?;
+
+        let txn = account1.sign_with_transaction_builder(factory.peer_to_peer(
+            Currency::XUS,
+            account2.address(),
+            4_000_000,
+        ));
+        client.submit(&txn)?;
+        client.wait_for_signed_transaction(&txn, None, None)?;
+
+        let events_fetcher = DiemEventsFetcher::new(ctx.url())?;
```

Comment:
considering this is only used in the runtime, why not instantiate it there? I guess the ctx?

Reply:
Not sure I follow what you're asking? What runtime? This is a construct that is only being tested and used in this single test.